pkg/webhook/risingwave_validating_webhook.go

```diff
@@ -328,6 +351,20 @@ func (v *RisingWaveValidatingWebhook) ValidateUpdate(ctx context.Context, oldObj
 	return v.validateUpdate(ctx, oldObj.(*risingwavev1alpha1.RisingWave), newObj.(*risingwavev1alpha1.RisingWave))
 }
 
+// groupTemplatePartiionExist checks if has been set inside the upgrade strategy.
+func groupTemplatePartiionExistAndIsString(groupTemplate *risingwavev1alpha1.RisingWaveComponentGroupTemplate) bool {
+	if groupTemplate.UpgradeStrategy.RollingUpdate == nil {
+		return false
+	}
+	if groupTemplate.UpgradeStrategy.RollingUpdate.Partition == nil {
+		return false
+	}
+	if groupTemplate.UpgradeStrategy.RollingUpdate.Partition.Type == intstr.Int {
+		return false
+	}
+	return true
```

Comment:
Check the type directly.

```suggestion
	return groupTemplate.UpgradeStrategy.RollingUpdate.Partition.Type == intstr.String
```